File: app/src/main/java/com/rabtman/acgclub/mvp/contract/ScheduleNewContract.java
package com.rabtman.acgclub.mvp.contract;

import com.rabtman.acgclub.mvp.model.jsoup.ScheduleNew;
import com.rabtman.common.base.mvp.IModel;
import com.rabtman.common.base.mvp.IView;
import io.reactivex.Flowable;

/**
 * @author Rabtman
 */
public interface ScheduleNewContract {

  interface View extends IView {

    void showScheduleNew(ScheduleNew scheduleNew);
  }

  interface Model extends IModel {

    Flowable<ScheduleNew> getScheduleNew(String url);
  }
}

File: component-acgschedule/src/main/java/com/rabtman/acgschedule/mvp/contract/ScheduleNewContract.kt
package com.rabtman.acgschedule.mvp.contract

import com.rabtman.acgschedule.mvp.model.jsoup.ScheduleNew
import com.rabtman.common.base.mvp.IModel
import com.rabtman.common.base.mvp.IView
import io.reactivex.Flowable

/**
 * @author Rabtman
 */
interface ScheduleNewContract {
    interface View : IView {
        fun showScheduleNew(scheduleNew: ScheduleNew?)
    }

    interface Model : IModel {
        fun getScheduleNew(url: String): Flowable<ScheduleNew>
    }
}